src/V3Const.cpp

```diff
@@ -853,7 +853,8 @@ class ConstBitOpTreeVisitor final : public VNVisitorConst {
         // of the masking node e.g. by the "AND with all ones" rule. If the result width happens
         // to be 1, we still need to ensure the AstAnd is not dropped, so use a wider mask in this
         // special case.
-        const int maskWidth = resultWidth == 1 ? VL_IDATASIZE : resultWidth;
+        const int maskWidth
+            = std::max(resultp->width(), resultWidth == 1 ? VL_IDATASIZE : resultWidth);
```

Comment:
If simplify to `maskWidth = resultp->width();` , t_const_bitoptree_bug3096 fails.
Until all width becomes correct, this is safer.